app/controllers/mappings_controller.rb

```diff
@@ -15,21 +16,29 @@ def create
     @mapping = Mapping.new(mapping_params)
 
     if @mapping.save_and_set_facls
-      flash[:success] = @mapping.save_message
+      flash[:success] = 'Mapping successfully created.'
       redirect_to action: :index
     else
-      flash[:warning] = 'Unable to create new mapping. ' + @mapping.save_message
-      redirect_to new_mapping_path, locals: params
+      flash[:warning] = 'Unable to create new mapping. ' + @mapping.errors.full_messages.join(' ')
```

Comment:
When doing render below, you should use `flash.now` instead of `flash`:

https://guides.rubyonrails.org/v4.2/action_controller_overview.html#the-flash (see 5.2.1 flash.now)

>By default, adding values to the flash will make them available to the next request, but sometimes you may want to access those values in the same request. For example, if the create action fails to save a resource and you render the new template directly, that's not going to result in a new request, but you may still want to display a message using the flash. To do this, you can use flash.now in the same way you use the normal flash:

Note the `flash.now` is special in the controller. This ensures when rendering the view `flash` is set so this works whether `flash` or `flash.now` was used in the controller:

https://github.com/OSC/shiny_admin/blob/46261c7d18decbc60f9be84ec67855ef3baecc19/app/views/layouts/application.html.erb#L55-L63